### ISSUE TITLE
Playback does not work in Safari

### ISSUE THREAD

**Issue by WikeDuivenvoorden** [DanielKoomen/WebApp#67]
Playing music is not working on Iphone. I tried using Safari and Firefox. All other functionalities work fine...

**Comment by Derkades** [comment]
Could you visit https://chrisguttandin.github.io/mse-tests/ and try to play the "Opus inside of a WebM container" file? Does it work? If not, what does the website say below the audio controls?

**Comment by Derkades** [comment]
If webm doesn't work, I guess I'll have to transcode all files twice, in the proprietary [CAF](https://en.wikipedia.org/wiki/Core_Audio_Format) audio format for iOS and opus/webm for everyone else.

**Comment by WikeDuivenvoorden** [comment]
> Could you visit https://chrisguttandin.github.io/mse-tests/ and try to play the "Opus inside of a WebM container" file? Does it work? If not, what does the website say below the audio controls?
> 
> 
> 
> The music player uses the widely used "opus" audio format, in ogg container (but could switch to webm if necessary). It has been supported since 2014 by Chromium and since 2012 by Firefox, but support [by Apple still seems lacking](https://caniuse.com/?search=opus) even though [they claim it should work](https://stackoverflow.com/questions/70143421/webm-and-opus-in-safari).
> 
> 
> 
> It'll be the same for all browsers on iOS, since Apple forces all browsers in the App Store to use the same WebKit rendering engine used by Safari.

None of the files worked. It also was strangely formatted, see pictures for both Firefox and Safari (asking for desktop version did nothing)

![image](https://user-images.githubusercontent.com/71707262/195989110-8de7428f-cb1d-4b19-b014-8fcd4c6cf680.png)

![image](https://user-images.githubusercontent.com/71707262/195989121-25d8fcd8-5423-4394-9e65-27ed442c0054.png)

**Comment by Derkades** [comment]
Perhaps safari has no native media controls? Or it could be blocking all audio elements for some reason

**Comment by Derkades** [comment]
I'll just switch from opus/ogg to opus/webm, we'll see what happens. Would be nice to have iOS suppport.

**Comment by WikeDuivenvoorden** [comment]
Don't know if this helps but I found this site (https://www.learningcontainer.com/sample-audio-file/) and the first mp3 file worked!

It also showed up in the control panel: 
![image](https://user-images.githubusercontent.com/71707262/195990036-f463add9-71e3-4bbd-aa4c-1343a83690d3.jpeg)

**Comment by Derkades** [comment]
Good to know `mp3/mpeg` works, that'll be a last resort if `opus/caf` doesn't work (mp3 is much larger for the same quality, or much lower quality for the same size)

**Comment by Derkades** [comment]
the change has been pushed to "production", would you mind trying again?

**Comment by WikeDuivenvoorden** [comment]
Tried again, but still no sound.

**Comment by Derkades** [comment]
Unfortunate. Opus/CAF up next, then

**Comment by Derkades** [comment]
It took a few hours but I finally managed to get macOS running in qemu/kvm for testing! I found that specifically the combination of AAC audio, in MP4 container, with `audio/mp4` mimetype seems to work. Hopefully it also works for iOS Safari, there's no way for me to test on iOS.

And indeed, `opus/webm` definitely does not work, not even on latest Safari. Even though Apple claims to the press it works since [macOS Big Sur](https://9to5mac.com/2021/02/18/apple-adds-webm-video-playback-support-to-safari-with-macos-big-sur-11-3/) and [iOS 15](https://9to5mac.com/2021/08/10/apple-adding-webm-audio-codec-support-to-safari-with-ios-15/)

**Comment by Derkades** [comment]
Pushed now, please test :-)

**Comment by WikeDuivenvoorden** [comment]
IT WORKS! :D

**Comment by WikeDuivenvoorden** [comment]
Oops I was a bit too fast. It plays music now, that's good. However after one song the next one doesn't start automatically, but this happens only after i lock my phone.

**Comment by Derkades** [comment]
It is probably because of Safari blocking autoplay by default. Like other browsers, there should be an option to allow autoplay for specific sites
